grommet/grommet
# [TS] Migrate MaskedInput tests to .tsx

What does this PR do?
This PR migrates MaskedInput component tests from *.js to *.tsx to allow ts-jest to catch type-related errors in the unit testing pipeline. Changes made:

1. Instead of passing document to getByText, we now use screen.getByText('aa'). The screen object is a globally available helper from @testing-library/react and doesn't require a container. This avoids the type mismatch error, as screen will search the rendered DOM.
2. optionButton Check: Since .closest('button') can potentially return null, I’ve added a check to ensure it exists before proceeding.

Where should the reviewer start?
Ensure all tests are currently passing

What testing has been done on this PR?
Manually ran and passed MaskedInput component tests locally.
CircleCI tests have passed.

How should this be manually tested?
Modified tests behave identically as before.

Do Jest tests follow these best practices?

Any background context you want to provide?

What are the relevant issues?
#5599 

Screenshots (if appropriate)

Do the grommet docs need to be updated?
no

Should this PR be mentioned in the release notes?
no

Is this change backwards compatible or is it a breaking change?
compatible
